packages/fungible-token/bridge-fungible-token/src/main.sw

```diff
@@ -52,7 +57,8 @@ configurable {
 storage {
     asset_to_sub_id: StorageMap<AssetId, b256> = StorageMap {},
```

Comment:
I see this is only used in the withdrawal function of the bridge, wouldn't it make more sense to pass it as an argument to the `withdraw(to: b256, sub_id: SubId)` function and assert that the `msg_asset_id() == AssetId::new(contract_id(), sub_id)`?

Reply:
This would imply that callers have to be aware of the `sub_id` of the asset they are withdrawing. There has been a lot of discussion around this and I think we have not yet reached an agreement, as you said, it might be that the `sub_id` might be available as part of the message in the future. For the time being, and given that block space remains cheap, the consensus is that we keep the mapping of asset_id <-> sub_id in storage so that callers do not need to be aware of the sub_id.

I do not have a strong opinion on this though.

Reply:
I 've also refactored the use of `b256` types and replaced it with SubId and AssetId where needed